libraries/botbuilder-ai/tests/qnaMakerDialog.test.js

```diff
@@ -242,4 +251,201 @@ describe('QnAMakerDialog', function () {
+            const dm = new DialogManager();
+            dm.conversationState = convoState;
+            const cardNoMatchText = 'Not helpful.';
+            const qnaDialog = new QnAMakerDialog(
+                kbId,
+                endpointKey,
+                HOSTNAME,
+                undefined,
+                undefined,
+                (suggestionsList, noMatchingQuestionsText) => {
+                    ok(suggestionsList);
+                    strictEqual(suggestionsList.length, 3);
+                    strictEqual(noMatchingQuestionsText, cardNoMatchText);
+                    return MessageFactory.suggestedActions(suggestionsList, noMatchingQuestionsText);
+                },
+                cardNoMatchText
+            );
+
+            dm.rootDialog = qnaDialog;
+            const adapter = new TestAdapter((turnContext) => {
+                return dm.onTurn(turnContext);
+            });
+
+            await adapter
+                .send('QnaMaker_TopNAnswer.json')
+                .assertReply((reply) => {
+                    strictEqual(reply.type, ActivityTypes.Message);
+                    strictEqual(reply.suggestedActions.actions.length, 3);
+                    strictEqual(reply.text, cardNoMatchText);
+                })
+                .startTest();
+        });
+
+        it('should error if suggestionsActivityFactory is passed in with falsy cardNoMatchText', () => {
+            const kbId = 'dummyKbId';
+            const endpointKey = 'dummyEndpointKey';
+            throws(
+                () => new QnAMakerDialog(kbId, endpointKey, HOSTNAME, undefined, undefined, (_) => {}, undefined),
+                new TypeError('cardNoMatchText is required when using the suggestionsActivityFactory.')
+            );
+        });
+
+        it('should error if suggestionsActivityFactory returns void', async () => {
+            const kbId = 'dummyKbId';
+            const endpointKey = 'dummyEndpointKey';
+            const convoState = new ConversationState(new MemoryStorage());
+            const dm = new DialogManager();
+            dm.conversationState = convoState;
+            const cardNoMatchText = 'Not helpful.';
+            const qnaDialog = new QnAMakerDialog(
+                kbId,
+                endpointKey,
+                HOSTNAME,
+                undefined,
+                undefined,
+                (suggestionsList, noMatchingQuestionsText) => {
+                    ok(suggestionsList);
+                    strictEqual(suggestionsList.length, 3);
+                    strictEqual(noMatchingQuestionsText, cardNoMatchText);
+                },
+                cardNoMatchText
+            );
+
+            dm.rootDialog = qnaDialog;
+            const adapter = new TestAdapter((turnContext) => {
+                return dm.onTurn(turnContext);
+            });
+
+            await adapter
+                .send('QnaMaker_TopNAnswer.json')
+                .catch((reason) => {
```

Comment:
This could either be `assert.rejects` or `.catch(reason => assert.deepStrictEqual(reason, {name: 'DialogContextError', message: 'Activity with ...'}))`

Reply:
Too close to code complete, I used `throws` earlier in this PR but neglected using `rejects` here. I'll clean this up.